Cabal-tests/tests/HackageTests.hs

```diff
@@ -63,23 +64,15 @@ import Data.TreeDiff.Pretty          (ansiWlEditExprCompact)
 parseIndex :: (Monoid a, NFData a) => (FilePath -> Bool)
            -> (FilePath -> B.ByteString -> IO a) -> IO a
 parseIndex predicate action = do
-    cabalDir   <- getAppUserDataDirectory "cabal"
-    configPath <- getCabalConfigPath cabalDir
+    configPath <- getConfigFilePath
     cfg        <- B.readFile configPath
     cfgFields  <- either (fail . show) pure $ Parsec.readFields cfg
+    repoCache  <- case lookupInConfig "remote-repo-cache" cfgFields of
+                    []        -> defaultCacheDir   -- Default
```

Comment:
this is change is good, and could be merged without xdg stuff.

Reply:
Alright, I'll prepare a separate PR.